### PR TITLE
ITS FHR: Changes in DeadChipPos and AliveChipPos plots

### PR DESCRIPTION
**Pull request by eeremenk**
DeadChipPos and AliveChipPos for z and phi coordinates implemented for IB and OB. Solution works for the IB, but there is still a binning problem for the OB. 
Layout with the DeadChipPos plots: https://qcg-test.cern.ch/?page=layoutShow&layoutId=63d918eaeb006ed9dd37b9c9

### CODE REVIEW

File: Modules/ITS/include/ITS/ITSFhrTask.h [hunk 1]
```diff
@@ -131,6 +133,9 @@ class ITSFhrTask final : public TaskInterface
   static constexpr int nbinsphiIB = 16;
   int mEtabins = 130;
   int mPhibins = 240;
+
+  const float mLength[NLayer] = { 27.1, 27.1, 27.1, 84.3, 84.3, 147.5, 147.5 };  //Sensetive area length  
+
   float etabinsIB[3][nbinsetaIB + 1] = { { -2.34442, -2.06375, -1.67756, -1.0732, 0., 1.0732, 1.67756, 2.06375, 2.34442, 2.6 }, { -2.05399, -1.77866, -1.40621, -0.851972, 0., 0.851972, 1.40621, 1.77866, 2.05399, 2.4 }, { -1.8412, -1.57209, -1.21477, -0.707629, 0., 0.707629, 1.21477, 1.57209, 1.8412, 2.1 } };
   float phibinsIB[3][nbinsphiIB + 1] = { { -2.86662, -2.34302, -1.81942, -1.29582, -0.772222, -0.248623, 0.203977, 0.274976, 0.596676, 0.798574, 0.989375, 1.32217, 1.38207, 1.84577, 2.36937, 2.89297, 3.2 }, { -2.93762, -2.54492, -2.15222, -1.75952, -1.36682, -0.974121, -0.581422, -0.188722, 0.159761, 0.47392, 0.78808, 1.10224, 1.77477, 2.16747, 2.56017, 2.95287, 3.2 }, { -2.98183, -2.66767, -2.35351, -2.03935, -1.72519, -1.41104, -1.09688, -0.782717, -0.468558, -0.154398, 1.4164, 1.73056, 2.04472, 2.35888, 2.67304, 2.98719, 3.2 } };
```
Comment (IsakovAD) [inline Modules/ITS/include/ITS/ITSFhrTask.h:140]: Delete etaBins and phiBins arrays if they are not used anymore

File: Modules/ITS/src/ITSFhrTask.cxx [hunk 2]
```diff
@@ -307,22 +318,10 @@ void ITSFhrTask::createOccupancyPlots() // create general plots like error, trig
...
-      mAliveChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dAliveChipPos", mLayer, mLayer), Form("AliveChipPos on Layer %d", mLayer), 28, etabinsOB4, 120, phibinsOB4);
-    } else if (mLayer == 5) {
-      mDeadChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dDeadChipPos", mLayer, mLayer), Form("DeadChipPos on Layer %d", mLayer), 49, etabinsOB5, 168, phibinsOB5);
-      mAliveChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dAliveChipPos", mLayer, mLayer), Form("AliveChipPos on Layer %d", mLayer), 49, etabinsOB5, 168, phibinsOB5);
-    } else {
-      mDeadChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dDeadChipPos", mLayer, mLayer), Form("DeadChipPos on Layer %d", mLayer), 49, etabinsOB6, 192, phibinsOB6);
-      mAliveChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dAliveChipPos", mLayer, mLayer), Form("AliveChipPos on Layer %d", mLayer), 49, etabinsOB6, 192, phibinsOB6);
-    }
+    mDeadChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dDeadChipPos", mLayer, mLayer), Form("DeadChipPos on Layer %d", mLayer), nHicPerStave[mLayer]*2, -0.5 * mLength[mLayer], 0.5 * mLength[mLayer], NStaves[mLayer]*7, -180, 180);
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFhrTask.cxx:321]: I have fixed the binning for OB, it should be something like that (you can see new plots in your layout):

mDeadChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dDeadChipPos", mLayer, mLayer), Form("DeadChipPos on Layer %d", mLayer), nHicPerStave[mLayer]*7*0.5, -0.5 * mLength[mLayer], 0.5 * mLength[mLayer], NStaves[mLayer]*4, -180, 180)

File: Modules/ITS/src/ITSFhrTask.cxx [hunk 3]
```diff
@@ -655,12 +656,12 @@ void ITSFhrTask::monitorData(o2::framework::ProcessingContext& ctx)
       for (int ichip = 0; ichip < nChipsPerHic[mLayer]; ichip++) {
         mChipStaveOccupancy->SetBinContent(ichip + 1, istave + 1, mOccupancyLane[istave][ichip]);
         if (!mChipStat[istave][ichip]) {
-          mDeadChipPos->SetBinContent(mDeadChipPos->GetXaxis()->FindBin(mChipEta[istave][ichip] + 0.009), mDeadChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip] + 0.001), 1);
+          mDeadChipPos->SetBinContent(mDeadChipPos->GetXaxis()->FindBin(mChipZ[istave][ichip]), mDeadChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip]), 1);
           mTotalDeadChipPos->SetBinContent(mTotalDeadChipPos->GetXaxis()->FindBin(mChipEta[istave][ichip] + 0.009), mTotalDeadChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip] + 0.001), 1);
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFhrTask.cxx:660]: Could you also fix "TotalChips" distributions to phi vs Z?

File: Modules/ITS/src/ITSFhrTask.cxx [hunk 4]
```diff
@@ -282,11 +294,10 @@ void ITSFhrTask::createOccupancyPlots() // create general plots like error, trig
       }
     }
 
-    mDeadChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dDeadChipPos", mLayer, mLayer), Form("DeadChipPos on Layer %d", mLayer), nbinsetaIB, etabinsIB[mLayer], nbinsphiIB, phibinsIB[mLayer]);    // every nine chips have same phi
-    mAliveChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dAliveChipPos", mLayer, mLayer), Form("AliveChipPos on Layer %d", mLayer), nbinsetaIB, etabinsIB[mLayer], nbinsphiIB, phibinsIB[mLayer]); // every nine chips have same phi
+    mDeadChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dDeadChipPos", mLayer, mLayer), Form("DeadChipPos on Layer %d", mLayer), nChipsPerHic[mLayer], -0.5 * mLength[mLayer], 0.5 * mLength[mLayer], NStaves[mLayer], -180, 180); 
+    mAliveChipPos = new TH2D(Form("Occupancy/Layer%d/Layer%dAliveChipPos", mLayer, mLayer), Form("AliveChipPos on Layer %d", mLayer), nChipsPerHic[mLayer] , -0.5 * mLength[mLayer], 0.5 * mLength[mLayer], NStaves[mLayer], -180, 180);
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFhrTask.cxx:298]: Would it be possible to show statistics for this histograms? I am speaking about the block with RMS, mean and number of entries?

Comment (eeremenk) [reply Modules/ITS/src/ITSFhrTask.cxx:298]: I have added the ResetStats function on lines 706 -707 (so it does not count 0 inputs as entries), that worked while testing locally, but i can see in the layout that it did not work when you tested the code?

File: Modules/ITS/src/ITSFhrTask.cxx [hunk 5]
```diff
@@ -657,12 +648,12 @@ void ITSFhrTask::monitorData(o2::framework::ProcessingContext& ctx)
...
-          mTotalAliveChipPos->SetBinContent(mTotalAliveChipPos->GetXaxis()->FindBin(mChipEta[istave][ichip] + 0.009), mTotalAliveChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip] + 0.001), 1);
+          mAliveChipPos->SetBinContent(mAliveChipPos->GetXaxis()->FindBin(mChipZ[istave][ichip]), mAliveChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip]), 1);
+          mTotalAliveChipPos->SetBinContent(mTotalAliveChipPos->GetXaxis()->FindBin(mChipZ[istave][ichip]), mTotalAliveChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip]), 1);
           mDeadChipPos->SetBinContent(mDeadChipPos->GetXaxis()->FindBin(mChipZ[istave][ichip] ), mDeadChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip] ), 0);                     // not dead
-          mTotalDeadChipPos->SetBinContent(mTotalDeadChipPos->GetXaxis()->FindBin(mChipEta[istave][ichip] + 0.009), mTotalDeadChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip] + 0.001), 0); // not dead
+          mTotalDeadChipPos->SetBinContent(mTotalDeadChipPos->GetXaxis()->FindBin(mChipZ[istave][ichip]), mTotalDeadChipPos->GetYaxis()->FindBin(mChipPhi[istave][ichip]), 1); // change to 0 after testing! // not dead
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFhrTask.cxx:656]: Shouldn't it be 0?

**Review by IsakovAD — CHANGES_REQUESTED** [review]
Code is working fine. Just fix this small error and we can merge

**Comment by IsakovAD** [comment]
@iravasen the changes fine for me. Could we merge?

**Comment by iravasen** [comment]
Fine with me as well, sorry I missed your comment.

**Review by iravasen — APPROVED** [review]

**Pull request merged**